File: files/400-499/419/CH-419.py
import pandas as pd

path = "400-499/419/CH-419 Filter.xlsx"
input = pd.read_excel(path, usecols="B:C", nrows=5, skiprows=2)
test = pd.read_excel(path, usecols="G:H", nrows=2, skiprows=2)
test.columns = input.columns


result = input[input['Sales'] > input.groupby('Product')['Sales'].transform('mean')].reset_index(drop=True)
print(result.equals(test))
# [1] True

File: files/400-499/419/CH-419.R
library(tidyverse)
library(readxl)

path <- "400-499/419/CH-419 Filter.xlsx"
input <- read_excel(path, range = "B3:C7")
test <- read_excel(path, range = "G3:H5")
colnames(test) = colnames(input)

result = input %>%
  filter(Sales > mean(Sales), .by = Product)

all.equal(result, test)
# [1] TRUE
